### PR TITLE
Replace Python loop in HDUIndexTable.row_index() by numpy indexing

### PR DESCRIPTION
**Pull request by adonath**
This PR implements a fix for #922, by replacing the Python loop with numpy indexing.

@registerrier I put you as a reviewer to confirm, that this fixes or at least improves the issue.

### CODE REVIEW

File: gammapy/data/hdu_index_table.py [hunk 1]
```diff
@@ -222,20 +222,20 @@ def row_idx(self, obs_id, hdu_type=None, hdu_class=None):
...
-        # Using a Python loop and cached lists of stripped string columns is easiest here.
-        idx_list = []
+        selection = self['OBS_ID'] == obs_id
 
-        for idx in range(len(self)):
-            if self['OBS_ID'][idx] == obs_id:
-                if hdu_class and self._hdu_class_stripped[idx] == hdu_class:
-                    idx_list.append(idx)
+        if hdu_class:
+            hdu_class_stripped = np.array(self._hdu_class_stripped)
```
Comment (cdeil) [inline gammapy/data/hdu_index_table.py]: If you stick with this implementation: why have the Python list -> numpy array conversion in the lookup?
Make the cached property a numpy array?

Comment (cdeil) [inline gammapy/data/hdu_index_table.py]: If you stick with this implementation: why have the Python list -> numpy array conversion in the lookup?
Make the cached property a numpy array?

Comment (registerrier) [inline gammapy/data/hdu_index_table.py]: This works much better. And indeed self._hdu_class_stripped could directly be a numpy array.

**Comment by adonath** [comment]
For now I decided to keep this implementation using numpy indexing, because we have to refactor the `HDUIndexTable` class anyway (see #946). Once Travis-CI passes I'll merge.

**Pull request merged**